src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.Encrypt.cs

```diff
@@ -22,6 +22,13 @@ internal sealed partial class PkcsPalOpenSsl : PkcsPal
         {
             int status;
 
+            if (originatorCerts.Count != 0)
+            {
+                // OpenSSL doesn't support adding certificates to a freshly created EnvelopedCMS (CMS_ContentInfo inside OpenSSL)
+                // these are used normally for KeyAgreement, but OpenSSL doesn't have support for it so we can't add these.
+                throw new PlatformNotSupportedException(SR.Cryptography_Cms_AddOriginatorCertsPlatformNotSupported);
```

Comment:
Since we're giving up and saying we're done trying, this should be codified in a test.

https://github.com/dotnet/corefx/blob/32b96925efce0dde416103862d0f37877918ec94/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs has some good examples of this:

* `SSLBackendNotSupported_Callback_ThrowsPlatformNotSupportedException` runs only when `BackendDoesNotSupportCustomCertificateHandling`
* `NoCallback_RevokedCertificate_RevocationChecking_Fails` runs only when `BackendSupportsCustomCertificateHandling`
* Both of those properties are defined at the bottom of the file, one as the simple negation of the other (though using expression-bodied members, in case you have yet to see that syntax and it throws you off).